contracts/Kernel.sol

```diff
@@ -268,7 +278,7 @@ contract Kernel is Factory {
             }
         }
         if (!status) {
-            err = 4;
```

Comment:
Would use an enum to specify error codes.

Reply:
I've made this into its own issue.